File: api/clients/src/main/java/com/foodagram/clients/content/dto/ReportCreationDto.java
package com.foodagram.clients.content.dto;

import com.foodagram.clients.content.enums.ReportReason;
import com.foodagram.clients.content.enums.ReportType;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ReportCreationDto {
    private UUID reporterId;
    private String reporterUsername;
    private ReportType reportType;
    private UUID reportedEntityId;
    private ReportReason reason;
    private String additionalNotes;
}

File: admin-panel/foodagram-admin-panel/src/models/content/dto/ReportCreationDto.ts
import {ReportReason} from "./ReportReason.ts";
import {ReportType} from "./ReportType.ts";

export interface ReportCreationDto {
    reportType: ReportType;
    reportedEntityId: string;
    reporterUsername: string;
    reason: ReportReason;
    additionalNotes?: string;
}



